### ISSUE TITLE
[DI] Confusing error message when an argument can't be autowired

### ISSUE THREAD

**Issue by javiereguiluz** [symfony/symfony#22848]
| Q                | A
| ---------------- | -----
| Bug report?      | yes?
| Feature request? | no
| BC Break report? | no
| Symfony version  | 3.3

In https://github.com/symfony/symfony-demo/pull/562 we're updating the Symfony Demo app to 3.3.0 RC1 and all the new DI features. During the upgrade I suffered lots of errors ... but the error messages were perfect. Examples:

```
AutowiringFailedException in AutowirePass.php line 271:
Cannot autowire service "AppBundle\EventListener\CommentNotificationSubscriber":
argument "$sender" of method "__construct()" must have a type-hint or be given a
value explicitly.

AutowiringFailedException in AutowirePass.php line 271: Cannot autowire service
"AppBundle\EventListener\RedirectToPreferredLocaleSubscriber": argument
"$locales" of method "__construct()" must have a type-hint or be given a value
explicitly.
```

However, when I removed this original service config for our Twig extension:

```yaml
app.twig.app_extension:
    public:    false
    class:     AppBundle\Twig\AppExtension
    arguments: ['@markdown', '%app_locales%']
    tags:
        - { name: twig.extension }
```

I saw this confusing error:

```
FatalThrowableError in AppExtension.php line 42:
Type error: Too few arguments to function
AppBundle\Twig\AppExtension::__construct(), 0 passed in /Users/javier/sf
/symfony-demo/var/cache/dev/appDevDebugProjectContainer.php on line 3565 and
exactly 2 expected
```

This was fixed when I added this:

```yaml
AppBundle\Twig\AppExtension:
    arguments:
        $locales: '%app_locales%'
```

Why did I see a different error message in this case? Thanks!

**Comment by weaverryan** [comment]
I've also gotten this exact bad error. I'll check into it!

**Comment by javiereguiluz** [comment]
Another example of this confusing error message. I'm refactoring the console commands to use DI instead of extending from ContainerAware. I added this:

```php
    public function __construct(EntityManagerInterface $em)
    {
        parent::__construct();

        $this->entityManager = $em;
    }
```

But somehow PHPStorm didn't import the `EntityManagerInterface` class automatically, and I saw this error:

```
Fatal error: Uncaught Symfony\Component\Debug\Exception\FatalThrowableError:
Type error: Too few arguments to function
AppBundle\Command\DeleteUserCommand::__construct(), 0 passed in symfony-
demo/var/cache/dev/appDevDebugProjectContainer.php on line 685 and exactly 1
expected in symfony-demo/src/AppBundle/Command/DeleteUserCommand.php:44

Stack trace:
#0 symfony-demo/var/cache/dev/appDevDebugProjectContainer.php(685): AppBundle\Command\DeleteUserCommand->__construct()
#1 symfony-demo/vendor/symfony/symfony/src/Symfony/Component/DependencyInjection/Container.php(341): appDevDebugProjectContainer->getConsole_Command_AppbundleCommandDeleteusercommandService()
#2 symfony-demo/vendor/symfony/symfony/src/Symfony/Bundle/FrameworkBundle/Console/Application.php(141): Symfony\Component\DependencyInjection\Container->get('console.command...')
#3 symfony-demo/vendor/symfony/symfony/src/Symfony/Bundle/FrameworkBundle/Console/Applicatio in symfony-demo/src/AppBundle/Command/DeleteUserCommand.php on line 44
```

**Comment by weaverryan** [comment]
Fix in #22857. This affects any service that's inlined - i.e. any service that is only be referenced by only one other service. For example, the Command class is only referenced by the Command `Application`, and so autowiring errors for the Command class are swallowed (and you get this bad error instead).
